mod_form.php

```diff
@@ -198,7 +199,13 @@ public function get_data() {
             !empty($data->guestlinkenabled)
             && \mod_bigbluebuttonbn\locallib\config::get('participant_guest_requires_access_code')
         ) {
-            $data->guestpass = bigbluebuttonbn_generate_access_code();
+
+            // Check current guestpass and see if its empty. If there isno
+            // value, generate and set it, otherwise leave it alone.
+            if (isset($data->instance)) {
+                $currentguestpass = $DB->get_field('bigbluebuttonbn', 'guestpass', ['id' => $data->instance]);
```

Comment:
Wouldn't $data->guestpass already be set and we simply need to not overwrite it unless its empty?

Reply:
I retested and it seems looks like the $data->guestpass is not set/provided, even though the value is set against the DB record
```
stdClass Object
(
    [type] => 0
    [name] => BBB
    [introeditor] => Array
        (
            [text] => 
            [format] => 1
            [itemid] => 955462412
        )

    [showdescription] => 0
    [welcome] => 
    [voicebridge] => 0
    [userlimit] => 0
    [record] => 1
    [recordallfromstart] => 0
    [recordhidebutton] => 0
    [muteonstart] => 0
    [recordings_html] => 1
    [recordings_deleted] => 1
    [recordings_imported] => 0
    [recordings_preview] => 1
    [guestlinkenabled] => 1
    [mform_isexpanded_id_permissions] => 1
    [participants] => [{"selectiontype":"all","selectionid":"all","role":"viewer"},{"selectiontype":"user","selectionid":"2","role":"moderator"}]
    [openingtime] => 0
    [closingtime] => 0
    [accessmodaltext] => 
    [visible] => 1
    [visibleoncoursepage] => 1
    [cmidnumber] => 
    [groupmode] => 0
    [groupingid] => 0
    [availabilityconditionsjson] => {"op":"&","c":[],"showc":[]}
    [completionunlocked] => 0
    [completion] => 1
    [completionexpected] => 0
    [tags] => Array
        (
        )

    [course] => 2
    [coursemodule] => 2
    [section] => 0
    [module] => 4
    [modulename] => bigbluebuttonbn
    [instance] => 1
    [add] => 0
    [update] => 2
    [return] => 1
    [sr] => 0
    [competencies] => Array
        (
        )

    [competency_rule] => 0
    [submitbutton] => Save and display
)

```

Reply:
Potentially another way to resolve this is to make the guestpass field available in the form itself (so it's savable in both locations)

Reply:
I'm not too fussed either way, I'm torn being doing it right and wasting more time